File: web/backend/models/patient_profile.py
from datetime import date, datetime
from typing import Optional

from pydantic import BaseModel, Field


class PatientProfileCreate(BaseModel):
    name: str = Field(min_length=1, max_length=100)
    relationship: str
    gender: Optional[str] = None
    birth_date: Optional[date] = None
    diagnosis_date: Optional[date] = None
    vitiligo_type: Optional[str] = None
    notes: Optional[str] = None


class PatientProfileUpdate(BaseModel):
    name: Optional[str] = Field(default=None, min_length=1, max_length=100)
    relationship: Optional[str] = None
    gender: Optional[str] = None
    birth_date: Optional[date] = None
    diagnosis_date: Optional[date] = None
    vitiligo_type: Optional[str] = None
    notes: Optional[str] = None


class PatientProfileResponse(BaseModel):
    id: int
    name: str
    relationship: str
    gender: Optional[str] = None
    birth_date: Optional[date] = None
    diagnosis_date: Optional[date] = None
    vitiligo_type: Optional[str] = None
    notes: Optional[str] = None
    is_self: bool
    created_at: datetime

    class Config:
        from_attributes = True


class ModuleDefaults(BaseModel):
    tracker_profile_id: Optional[int] = None
    report_profile_id: Optional[int] = None
    diary_profile_id: Optional[int] = None

File: web/app/src/api/patient-profile.ts
import apiClient from './client'

export interface PatientProfile {
  id: number
  name: string
  relationship: string
  gender: string | null
  birth_date: string | null
  diagnosis_date: string | null
  vitiligo_type: string | null
  notes: string | null
  is_self: boolean
  created_at: string
}

export interface ModuleDefaults {
  tracker_profile_id: number | null
  report_profile_id: number | null
  diary_profile_id: number | null
}

export const patientProfileApi = {
  async list(): Promise<PatientProfile[]> {
    const { data } = await apiClient.get('/patient-profiles/')
    return data
  },

  async create(profileData: Partial<PatientProfile>): Promise<PatientProfile> {
    const { data } = await apiClient.post('/patient-profiles/', profileData)
    return data
  },

  async update(id: number, profileData: Partial<PatientProfile>): Promise<PatientProfile> {
    const { data } = await apiClient.put(`/patient-profiles/${id}`, profileData)
    return data
  },

  async delete(id: number): Promise<void> {
    await apiClient.delete(`/patient-profiles/${id}`)
  },

  async getModuleDefaults(): Promise<ModuleDefaults> {
    const { data } = await apiClient.get('/user/module-defaults')
    return data
  },

  async setModuleDefaults(defaultsData: Partial<ModuleDefaults>): Promise<ModuleDefaults> {
    const { data } = await apiClient.put('/user/module-defaults', defaultsData)
    return data
  }
}
